### PR TITLE
Provide temporary sysprop for disabling HFJ 5.0 query optimization

### PR DESCRIPTION
**Pull request by jamesagnew**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+	/**
+	 * Non instantiable
+	 */
+	private JpaSystemProperties() {
+	}
+
+	public static void updateSettingsBasedOnCurrentSystemProperties() {
+		ourOptmizeSingleElementInExpression = !("true".equals(System.getProperty(HFJ_FORCE_IN_CLAUSES_HF_50)));
+		if (ourOptmizeSingleElementInExpression) {
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java:24]: I believe this if needs to be reversed.  if (!ourO...

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchNoFtTest.java [hunk 2]
```diff
@@ -4730,6 +4732,48 @@ public void testCircularReferencesDontBreakRevIncludes() {
...
+			JpaSystemProperties.updateSettingsBasedOnCurrentSystemProperties();
+
+			myCaptureQueriesListener.clear();
+			SearchParameterMap map = new SearchParameterMap();
+			map.setLoadSynchronous(true);
+			map.add(AuditEvent.SP_ENTITY, new ReferenceParam(patientId));
+			IBundleProvider outcome = myAuditEventDao.search(map);
+			assertThat(toUnqualifiedVersionlessIdValues(outcome), contains(auditId));
+			myCaptureQueriesListener.logSelectQueriesForCurrentThread();
+			assertThat(myCaptureQueriesListener.getSelectQueriesForCurrentThread().get(0).getSql(true, true), containsString("myresource1_.TARGET_RESOURCE_ID in "));
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchNoFtTest.java:4770]: I love these tests.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package ca.uhn.fhir.jpa.dao.predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JpaSystemProperties {
+
+	private static final Logger ourLog = LoggerFactory.getLogger(JpaSystemProperties.class);
+	public static final String HFJ_FORCE_IN_CLAUSES_HF_50 = "hfj_force_in_clauses_hf50";
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java:9]: How can we ensure we back this out a year from now.  Is there some build-breaking reminder we could set for ourselves?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/predicate/JpaSystemProperties.java:24]: you are absolutely right

**Pull request merged**